validation/src/main/java/io/micronaut/validation/requires/RequiresConfigPropertiesVisitor.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    @Override
+    public BeanElement visitBeanElement(@NonNull BeanElement beanElement,
+                                        @NonNull VisitorContext visitorContext) {
+        final AnnotationValue<Requires> requiresAnnotation = beanElement.getAnnotation(Requires.class);
+        if (requiresAnnotation != null &&
+                (requiresAnnotation.contains(RequiresCondition.MEMBER_CONFIGURATION_PROPERTIES) ||
+                        requiresAnnotation.contains(RequiresCondition.MEMBER_CONFIGURATION_PROPERTY))) {
+            requiresConfigPropsBeans.add(beanElement);
+        }
+
+        AnnotationValue<ConfigurationProperties> configPropertiesAnnotation = beanElement.getAnnotation(ConfigurationProperties.class);
+        if (configPropertiesAnnotation != null) {
+            configPropertiesBeans.add(beanElement);
+        }
+
+        return beanElement;
+    }
+
+    @Override
+    public void finish(VisitorContext visitorContext) {
+        for (BeanElement conditionalBean : requiresConfigPropsBeans) {
+            AnnotationValue<Requires> requiresAnnotation = conditionalBean.getAnnotation(Requires.class);
+            if (requiresAnnotation == null || metadataBuilder == null) {
+                return;
+            }
+
+            String configProperty = requiresAnnotation.stringValue(RequiresCondition.MEMBER_CONFIGURATION_PROPERTY).orElse(null);
+            String configPropertiesClass = requiresAnnotation.stringValue(RequiresCondition.MEMBER_CONFIGURATION_PROPERTIES).orElse(null);
+
+            if (configPropertiesClass == null && configProperty != null) {
+                fail("configProperty member of @Requires annotation can not be used without specifying configProperties class", visitorContext, conditionalBean);
+                return;
+            }
+
+            boolean isConfigPropertyBean = false;
+            if (configPropertiesClass != null) {
+                for (BeanElement configPropertiesBean: configPropertiesBeans) {
+                    String className = getClassName(configPropertiesBean);
+                    if (className.equals(configPropertiesClass)) {
+                        isConfigPropertyBean = true;
+                        break;
+                    }
+                }
+            }
+
+            if (!isConfigPropertyBean) {
+                fail("Specified class [" + configPropertiesClass + "] must be configuration properties bean", visitorContext, conditionalBean);
+                return;
+            }
+
+            configPropertiesClass = configPropertiesClass.replace("$", ".");
+            if (configProperty != null) {
+                boolean propertyPresent = false;
+                for (PropertyMetadata propertyMetadata: metadataBuilder.getProperties()) {
+                    if (propertyMetadata.getDeclaringType().equals(configPropertiesClass) && propertyMetadata.getName().equals(configProperty)) {
```

Comment:
I'm not sure this is sufficient with regards to Java visibility rules. What if the config property is package private and in a different package and therefore inaccessible? I think only public configuration properties should be considered.